Userland/Libraries/LibWeb/HTML/VideoTrackList.cpp

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2023, Tim Flynn <trflynn89@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibJS/Runtime/Realm.h>
+#include <LibJS/Runtime/VM.h>
+#include <LibWeb/Bindings/Intrinsics.h>
+#include <LibWeb/Bindings/VideoTrackListPrototype.h>
+#include <LibWeb/HTML/EventNames.h>
+#include <LibWeb/HTML/VideoTrackList.h>
+
+namespace Web::HTML {
+
+VideoTrackList::VideoTrackList(JS::Realm& realm)
+    : DOM::EventTarget(realm)
+    , m_video_tracks(realm.heap())
+{
+}
+
+JS::ThrowCompletionOr<void> VideoTrackList::initialize(JS::Realm& realm)
+{
+    MUST_OR_THROW_OOM(Base::initialize(realm));
+    set_prototype(&Bindings::ensure_web_prototype<Bindings::VideoTrackListPrototype>(realm, "VideoTrackList"));
+
+    return {};
+}
+
+// https://html.spec.whatwg.org/multipage/media.html#dom-tracklist-item
+JS::ThrowCompletionOr<Optional<JS::PropertyDescriptor>> VideoTrackList::internal_get_own_property(JS::PropertyKey const& property_name) const
+{
+    if (property_name.is_number()) {
+        if (auto index = property_name.as_number(); index < m_video_tracks.size()) {
+            JS::PropertyDescriptor descriptor;
+            descriptor.value = m_video_tracks.at(index);
+
+            return descriptor;
+        }
+    }
+
+    return Base::internal_get_own_property(property_name);
+}
+
+ErrorOr<void> VideoTrackList::add_track(Badge<HTMLMediaElement>, JS::NonnullGCPtr<VideoTrack> video_track)
+{
+    TRY(m_video_tracks.try_append(video_track));
+    video_track->set_video_track_list({}, this);
+
+    return {};
+}
+
+JS::GCPtr<VideoTrack> VideoTrackList::selected_track(Badge<HTMLMediaElement>) const
+{
+    auto it = m_video_tracks.find_if([&](auto const& video_track) {
+        return video_track->selected();
+    });
+
+    if (it == m_video_tracks.end())
+        return nullptr;
+    return *it;
+}
+
+void VideoTrackList::remove_all_tracks(Badge<HTMLMediaElement>)
+{
+    m_video_tracks.clear();
+}
+
+// https://html.spec.whatwg.org/multipage/media.html#dom-videotracklist-gettrackbyid
+JS::GCPtr<VideoTrack> VideoTrackList::get_track_by_id(StringView id) const
+{
+    auto it = m_video_tracks.find_if([&](auto const& video_track) {
```

Comment:
`// The AudioTrackList getTrackById(id) and VideoTrackList getTrackById(id) methods must return the first AudioTrack or VideoTrack object (respectively) in the AudioTrackList or VideoTrackList object (respectively) whose identifier is equal to the value of the id argument (in the natural order of the list, as defined above).` (wrapped at ~120 chars)